eyeball-im-util/src/vector/sort.rs

```diff
@@ -0,0 +1,613 @@
+use std::{
+    cmp::Ordering,
+    ops::{ControlFlow, Not},
+    pin::Pin,
+    task::{self, ready, Poll},
+};
+
+use eyeball_im::{Vector, VectorDiff};
+use futures_core::Stream;
+use pin_project_lite::pin_project;
+use smallvec::SmallVec;
+
+use super::{
+    VectorDiffContainer, VectorDiffContainerOps, VectorDiffContainerStreamBuffer,
+    VectorDiffContainerStreamElement,
+};
+
+type UnsortedIndex = usize;
```

Comment:
In the `Filter` adapter, the "index of the element in the input vector" is called the *original* index. Would you mind using the same name here?

Reply:
I don't believe the naming should be identical in all stream adapters. The semantics take the priority over the consistency in some cases, and I reckon this is one of the cases. What matters —to me— is that we have facing an index that represented the unsorted position of the value. Whether it is the original or not the original index isn't really important. Thoughts?

Reply:
Yes, the semantics are what's important definitely. I still think it's a clearer name. The input / original list could already be sorted ;)

I'd also be okay with input / output index naming (for all adapters that re-order or filter items), how do you feel about that option?

Reply:
Do you want all stream adapters to use the same terminology?